packages/flutter/lib/src/widgets/visibility.dart

```diff
@@ -0,0 +1,209 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'ticker_provider.dart';
+
+/// Whether to show or hide a child.
+///
+/// By default, the [visible] property controls whether the [child] is included
+/// in the subtree or not; when it is not [visible], the [replacement] child
```

Comment:
The replacement feature is not something I would expect to be done by a Visibility widget...
Do you expect that people will want to change maintainState and maintainSize for a Visibility widget dynamically? if not maybe it will be simpler to have the visibility widget only do show/hide and have a Replace widget for the replacement logic

Reply:
I wouldn't expect anything but `visible` to be changed dynamically.
`replacement` is there because it's not clear to me that `SizedBox.shrink` is always the right choice for the replacement (in the case of `maintainState` being false).